docs/blog/2024-10-03-nx-20-release.md

```diff
@@ -0,0 +1,75 @@
+---
+title: Announcing Nx 20
+slug: announcing-nx-20
+authors: [Mike Hartington]
+tags: [nx, release]
+cover_image: /blog/images/2024-10-03/nx-20-header.png
+---
+
+I know it's hard to believe but Nx 20 is here! There's a lot to cover in this release, so let's just dive right in.
+
+## `@nx` All The Things!
+
+If you've been around the Nx ecosystem for any amount of time, you're probably aware that our plugins were published originally under the `@nrwl` npm scope. Since version 16, we've provided backwards compatibility by reexporting `@nx` scoped packages from `@nrwl`. With Nx 20 though, we'll be officially dropping support for the `@nrwl` scoped packages and they will no longer be published to npm. For projects that still have the older packages in them, our [migration docs][rescope] can guide you to what the new packages are in order for you to update any impacted dependencies.
+
+## TypeScript Project References For Monorepos
+
+[**Project References**](https://www.typescriptlang.org/docs/handbook/project-references.html) allows TypeScript to process individual projects rather than the entire monorepo as whole. This is a huge improvement in terms of build times and developer experience. However, project references comes with a maintenance cost, as it is up to you to update these references. In a large monorepo, this cost is prohibitive.
+
+With Nx 20, when using `create-nx-workspace --preset=ts`, workspaces will use Project References out of the box. Nx removes the maintenance cost of this feature by automatically updating the references for you when you run `build` or `typecheck` targets. You can also run [`nx sync`](https://nx.dev/reference/nx-commands#sync) to update references, and `nx sync:check` in CI to validate the workspace.
+
+In addition to adopting Project References, we're also changing the way we link projects in a TypeScript monorepos. **Workspaces** is the standard way to link packages in monorepos. This is a feature that all modern package managers support, such as [npm](https://docs.npmjs.com/cli/using-npm/workspaces), [yarn](https://yarnpkg.com/features/workspaces), [pnpm](https://pnpm.io/workspaces), and [bun](https://bun.sh/docs/install/workspaces), so taking advantage of it is a no-brainer. Now we have a proper way to provide type information and have your packages available in a way developers are used to.
+
+Note that we've enabled Project References and Workspaces for the TS preset (i.e. `--preset=ts`), and we're working on extending this support for all other presets soon (Angular, React, Vue, Node). If you are an existing Nx user and want to create an empty workspace in the previous "integrated" style, you can use `create-nx-workspace --preset=apps`.
```

Comment:
I would add a paragraph here stating that:
- This initial implementation creates new workspaces with the project references style, but there is no migration path yet.
- The project generators will detect which style of project linking you're using and generate the corresponding configuration code.
- Something to convey the idea that this is the direction we intend to go in the future, and you can try it out with a new repository, but it has no immediate impact on existing workspaces.